QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeBase.java

```diff
@@ -0,0 +1,48 @@
+package org.team1540.quasarhelios;
+
+import ccre.channel.BooleanInputPoll;
+import ccre.ctrl.BooleanMixing;
+import ccre.holders.TuningContext;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+
+public abstract class AutonomousModeBase extends InstinctModeModule {
+	public AutonomousModeBase(String modeName) {
+		super(modeName);
+	}
+	
+	protected void drive(float distance) throws AutonomousModeOverException, InterruptedException {
+		float currentEncoder = DriveCode.leftEncoder.get();
+		DriveCode.allMotors.set(distance/distance);
+		waitUntilAtLeast(DriveCode.leftEncoder, currentEncoder + distance);
+		DriveCode.allMotors.set(0.0f);
+	}
+	
+	protected void turn(float degree) throws AutonomousModeOverException, InterruptedException {
+		DriveCode.rotate.set(degree/degree);
+		waitUntilAtLeast(HeadingSensor.yaw, degree);
```

Comment:
keep in mind that yaw is absolute (sort of), not relative. Is this method supposed to turn to a specific angle or turn by a specific angle? Either way, it shouldn't be implemented like this.